src/session.rs

```diff
@@ -270,13 +272,85 @@ impl Session {
     ///
     /// If `program` is not an absolute path, the `PATH` will be searched in an OS-defined way on
     /// the host.
-    pub fn raw_command<S: AsRef<OsStr>>(&self, program: S) -> Command<'_> {
+    pub fn raw_command<S: AsRef<OsStr>>(&self, program: S) -> Command<&'_ Self> {
```

Comment:
You can forward command and raw_command to to_command and to_raw_command

Reply:
Interestingly no, because there is no `AsRef<T> for &T`.

Reply:
Hmmm, that's unexpected

Reply:
Yep it also surprised me when I hit it. One of the reasons I'm always a little bit confused about using AsRef v Borrow v Deref.

Here AsRef might be the wrong choice because it allows "transverse" ref conversions e.g. `AsRef<OsStr> for str` and has more flexibility but I'm not sure that's needed.

`Borrow` and `Deref` are more limited but might be more suitable.

Reply:
In that case, maybe swap that to Deref, since reference/Arc/Rc etc all have deref implemented.